pkg/apis/componentconfig/v1alpha1/types.go

```diff
@@ -0,0 +1,64 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import "k8s.io/kubernetes/pkg/api/unversioned"
+
+type KubeProxyConfiguration struct {
+	unversioned.TypeMeta
+
+	// bindAddress is the IP address for the proxy server to serve on (set to 0.0.0.0 for all interfaces)
+	BindAddress string `json:"bindAddress"`
+	// cleanupIPTables
+	CleanupIPTables bool `json:"cleanupIPTables"`
+	// healthzBindAddress is the IP address for the health check server to serve on, defaulting to 127.0.0.1 (set to 0.0.0.0 for all interfaces)
+	HealthzBindAddress string `json:"healthzBindAddress"`
+	// healthzPort is the port to bind the health check server. Use 0 to disable.
+	HealthzPort int `json:"healthzPort"`
+	// hostnameOverride, if non-empty, will be used as the identity instead of the actual hostname.
+	HostnameOverride string `json:"hostnameOverride"`
+	// iptablesSyncPeriodSeconds is the period that iptables rules are refreshed (e.g. '5s', '1m', '2h22m').  Must be greater than 0.
+	IPTablesSyncePeriodSeconds int `json:"iptablesSyncPeriodSeconds"`
+	// kubeAPIBurst is the burst to use while talking with kubernetes apiserver
+	KubeAPIBurst int `json:"kubeAPIBurst"`
+	// kubeAPIQPS is the max QPS to use while talking with kubernetes apiserver
+	KubeAPIQPS int `json:"kubeAPIQPS"`
+	// kubeconfigPath is the path to the kubeconfig file with authorization information (the master location is set by the master flag).
+	KubeconfigPath string `json:"kubeconfigPath"`
```

Comment:
if paths are specified in config files, we need to decide how relative paths are resolved (currently it is resolved relative to the working directory of the process). Relative file references from config files should be resolved relative to the directory containing the config file. We came up with the following approach in OpenShift, where path resolution is an explicit operation invoked on a config object:

https://github.com/openshift/origin/blob/3cf7f3cccd5b7b95448568ba8e2b3c81cc6ab9af/pkg/cmd/server/api/helpers.go#L240-L268